src/http/StateFlags.h

```diff
@@ -16,7 +16,7 @@ class StateFlags
 {
 public:
     unsigned int front_end_https = 0; ///< send "Front-End-Https: On" header (off/on/auto=2)
-    bool keepalive = false;
+    bool keepalive = false; ///< whether to keep the connection persistent
```

Comment:
Is this the right default still? false is the default for HTTP/1.0 semantics, while implicit true would match HTTP/1.1 requirements. I suspect this would need a lot more code change so do not insist on it, but *if* we know for sure that this needs to change now would be a good time to add a TODO.

Reply:
There is a relatively complex/unobvious logic of flags.keepalive initialization in HttpStateData::sendRequest(). However, it is clear that 'keepalive' cannot have a single default value(covering all cases). For example, for (peer && peer->stats.n_keepalives_sent >= 10) precondition the default value should be 'false', but for (peer==null) precondition it should be 'true'. I think that the proper requirement should be that  "flags.keepalive must be initialized before its first usage". But we cannot do that by means of a boolean flag.

Reply:
Thank you.

Reply:
> I think that the proper requirement should be that "flags.keepalive must be initialized before its first usage". But we cannot do that by means of a boolean flag.

Exactly! _And_ a `false` value is a "safer" default because the wrong `true` default may stall some transactions (which is worse than performance implications of the wrong `false` default).